subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/results/PerformanceTestResult.java

```diff
@@ -37,6 +37,20 @@ public  PerformanceTestResult() {
         whereAmI = new Throwable();
     }
 
+    /**
+     * Returns true if checks is enabled.
+     *
+     * When checks is enabled, an exception is thrown immediately upon two kind of failure:
+     *
+     * <ul>
+     * <li>A performance test iteration fails.</li>
```

Comment:
Thanks for adding the doc.

I am just thinking now: Should we always fail in case of an exception? Also in the case of `hasChecks () == false`? If there is an exception, something is wrong and even if I just want to measure, the measuring probably won't work correctly.

Reply:
I agree. If you can make that change it would be great.

If you like, you can do that in a separate PR though and merge this one as it is now. It's already an improvement.